pilot/cmd/istioctl/gendeployment/cmd.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gendeployment
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultTag = "0.4.0"
```

Comment:
what is this/what is it for ? why is there a hardcoded default / it shouldn't we can't update that file for every release

Reply:
We need some default value, this was current when the PR was merged. If there's a value we can use to do the right thing without having to update this, then we should use that.

Reply:
can we always -X link time set the current value of $HUB/$TAG instead (in the makefile) ?
and if one builds with "go install" directly they would have an empty default

worst case I guess you could use "does-not-exist" or "notset" 
(or "master"?)